packages/next/build/polyfills/process.js

```diff
@@ -1 +1,4 @@
-module.exports = global.process || require('../../compiled/process')
+module.exports =
+  typeof global.process?.env === 'object'
```

Comment:
```suggestion
  global.process?.env && typeof global.process?.env === 'object'
```
Might be good to check for the case where `global.process.env` ends up `null` somehow.